### ISSUE TITLE
Shape.Rectangle(point, size) draws Circle in some cases

### ISSUE THREAD

**Issue by AndreyChernykh** [paperjs/paper.js#1245]
If we use ```Shape.Rectangle(point, size)``` with ```size = new Size(x, y);``` where both ```x``` and ```y``` are negative, a **Circle is drawn** instead of a Rectangle. 

Example: 
```js
var point = new Point(50, 50);
var size = new Size(-10, -10);
var shape = new Shape.Rectangle(point, size);
shape.strokeColor = 'black';
```

Could be checked here: 
http://paperjs.org/reference/shape/#shape-rectangle-point-size

**Comment by lehni** [comment]
This happens because you are defining a negative size, and due to this line of code:

```js
 this._radius.set(Size.min(this._radius, size.divide(2)));
```
If size is negative, then `size.divide(2)` will always end up being used for the radius.

I am not sure we should support negative sizes in Shape items, it wasn't really the intention at least to do so. What's your use case for negative dimensions?

**Comment by pdubroy** [comment]
I recently ran into a similar issue. In my case, it's a drawing application, and I implemented rectangle drawing that works in the usual way: one of the corners is set on `mousedown`, and the other corner is set on `mouseup`. While the mouse is still down, I adjust the width of the rectangle based on the difference between the current mouse position and the position of the `mousedown` event.

Naively, I expected that setting a negative width or height would just flip the rectangle along the appropriate axis. E.g., in the example given above, I'd expect a 10x10 rectangle whose bottom right corner is at 50, 50.

It's certainly not hard to work around, but the code is much simpler without having to special case this (and it seems to have a straightforward geometric interpretation).
